components/tidb_query_expr/src/impl_time.rs

```diff
@@ -218,6 +219,94 @@ pub fn make_date(ctx: &mut EvalContext, year: &Int, day: &Int) -> Result<Option<
     Ok(Some(ret))
 }
 
+pub fn make_time(
+    ctx: &mut EvalContext,
+    hour: &Int,
+    minute: &Int,
+    second: &Real,
+    fsp: i8,
+    isunsinged: bool,
+) -> Result<Option<Duration>> {
+    let mut hour = *hour;
```

Comment:
You can reuse [`Duration::new_from_parts()`](https://tikv.github.io/doc/tidb_query_datatype/codec/mysql/duration/struct.Duration.html#method.new_from_parts)